aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -116,6 +117,10 @@ inline dim3 SpatialSoftMax_getGridSize(
 
 const int max_threads = 1024;
 
+// The Nvidia Kepler GPUs have 48KB of shmem per SM. Future generations
+// have more. We use the minimum here to preserve some backwards compatibility.
+const int max_smem_per_block = 48 * 1024;
```

Comment:
Wouldn't it be better to query? Also, not sure if this statement is true for ROCm GPUs
cc: @jeffdaily

Reply:
Yep, let me see how we do it for other kernels and if there's no precedent I'll do a query and use a thread local storage to cache.

Reply:
Noticed that many other kernels use `at::cuda::getCurrentDeviceProperties()->sharedMemPerBlock` so I just added the same query.